### PR TITLE
Docker named volume

### PR DESCRIPTION
**Pull request by BulatSaif**
Signed-off-by: Bulat Saifullin <bulat@saifullin.ru>
changes:
1. added git ignore
2. added variable to configure named volume 
**UPD:**
3. update readme
4. move the loop over `hostvars` to jinja2 

The role is backward compatible.

### CODE REVIEW

File: ansible/roles/iroha-docker/templates/docker-compose.yml.j2 [hunk 1]
```diff
@@ -29,7 +29,7 @@ services:
 {% endfor %}
 {% endif %}
     volumes:
-      - iroha_block_store-{{ node.human_hostname }}:/tmp/block_store
+      - {{ iroha_volume_dir }}iroha_block_store-{{ node.human_hostname }}:/tmp/block_store
```
Comment (kogeler) [inline ansible/roles/iroha-docker/templates/docker-compose.yml.j2]: You maybe have to use conditional and "path" filter.

File: ansible/roles/iroha-docker/templates/docker-compose.yml.j2 [hunk 2]
```diff
@@ -62,19 +62,21 @@ services:
 {% endfor %}
 {% endif %}
     volumes:
-      - psql_storage-{{ iroha_inventory_human_hostname }}:/var/lib/postgresql/data
+      - {{ iroha_volume_dir }}psql_storage-{{ iroha_inventory_human_hostname }}:/var/lib/postgresql/data
```
Comment (kogeler) [inline ansible/roles/iroha-docker/templates/docker-compose.yml.j2]: You maybe have to use conditional and "path" filter.

File: ansible/roles/iroha-docker/templates/docker-compose.yml.j2 [hunk 3]
```diff
@@ -62,19 +62,21 @@ services:
...
     networks:
       - iroha-db-net
     restart: always
 {% if iroha_postgres_commands is defined %}
     command: {{ iroha_postgres_commands }}
 {% endif %}
 
+{% if iroha_volume_dir == "" %}
```
Comment (kogeler) [inline ansible/roles/iroha-docker/templates/docker-compose.yml.j2]:
```suggestion
{% if not iroha_volume_dir %}
```

File: ansible/roles/iroha-docker/templates/docker-compose.yml.j2 (hunk 1 above)
Comment (BulatSaif) [reply ansible/roles/iroha-docker/templates/docker-compose.yml.j2]: There are no `path` filter I used `dirname`, and put it to top of file

File: ansible/roles/iroha-docker/templates/docker-compose.yml.j2 (hunk 2 above)
Comment (BulatSaif) [reply ansible/roles/iroha-docker/templates/docker-compose.yml.j2]: There are no `path` filter I used `dirname`, and put it to top of file

File: ansible/ansible.cfg [hunk 4]
```diff
@@ -3,6 +3,7 @@
 
 [defaults]
 
+#callback_whitelist = profile_tasks
```
Comment (BulatSaif) [inline ansible/ansible.cfg]: Very good plugin for testing, it generate output like this :

```
PLAY RECAP *********************************************************************************************************************
localhost                  : ok=47   changed=14   unreachable=0    failed=0    skipped=20   rescued=0    ignored=0   

Wednesday 02 October 2019  18:27:21 +0300 (0:00:45.473)       0:01:22.126 ***** 
=============================================================================== 
iroha-docker : restart Iroha --------------------------------------------------------------------------------------------- 45.47s
Gathering Facts ----------------------------------------------------------------------------------------------------------- 9.36s
iroha-docker : Run Iroha -------------------------------------------------------------------------------------------------- 7.30s
iroha-docker : Build docker image for iroha-python util ------------------------------------------------------------------- 4.33s
iroha-docker : Generate peer keys ----------------------------------------------------------------------------------------- 2.83s
iroha-docker : Write 'iroha_peer_keys.yml' file --------------------------------------------------------------------------- 1.05s
iroha-docker : Check 'iroha_peer_keys.yml' file --------------------------------------------------------------------------- 0.97s
iroha-docker : Move priv peer keys ---------------------------------------------------------------------------------------- 0.96s
iroha-docker : Generate config files -------------------------------------------------------------------------------------- 0.94s
iroha-docker : Move pub peer keys ----------------------------------------------------------------------------------------- 0.90s
iroha-docker : Move genesis.block ----------------------------------------------------------------------------------------- 0.89s
iroha-docker : Create Docker network 2 ------------------------------------------------------------------------------------ 0.54s
iroha-docker : Copy iroha-python util ------------------------------------------------------------------------------------- 0.51s
iroha-docker : Create deploy dir ------------------------------------------------------------------------------------------ 0.48s
iroha-docker : Check iroha ------------------------------------------------------------------------------------------------ 0.43s
iroha-docker : Create deploy dir ------------------------------------------------------------------------------------------ 0.32s
iroha-docker : Copy iroha-python util Dockerfile -------------------------------------------------------------------------- 0.28s
iroha-docker : Generate Docker Compose file ------------------------------------------------------------------------------- 0.27s
iroha-docker : Generate genesis.block 2 ----------------------------------------------------------------------------------- 0.25s
iroha-docker : Generate hostnames (overlay/internal) ---------------------------------------------------------------------- 0.15s

```

**Pull request merged**